### ISSUE TITLE
Port GATK3 PR https://github.com/broadinstitute/gsa-unstable/pull/1389 "Changes to use the median rather than the second best likelihood for the NON_REF allele"

### ISSUE THREAD

**Issue by vruano** [broadinstitute/gatk#1950]
Create a PR in gatk to port the change once it has been merged into GATK3.
https://github.com/broadinstitute/gsa-unstable/pull/1389

**Comment by droazen** [comment]
@vruano @davidbenjamin Could one of you check whether this change was ever ported?

**Comment by davidbenjamin** [comment]
@droazen @vruano It was not ported to gATK4.

**Comment by droazen** [comment]
Tagging @vdauwera to ensure this change gets ported from GATK3.

**Comment by droazen** [comment]
Re-assigning to @asmirnov239

**Comment by droazen** [comment]
@davidbenjamin Can you comment here with the new location for the `ReadLikelihoods. updateNonRefAlleleLikelihoods()` method touched in the PR linked above? Our goal is to match current GATK3 behavior for both the old and new models.

**Comment by davidbenjamin** [comment]
@droazen I am more confident that this method was not ported (or deleted, I suppose).  It is used only once in gsa-unstable.  The GATK 3 method `HaplotypeCallerGenotypingEngine::prepareReadAlleleLikelihoodsForAnnotation` is identical to the GATK 4 method `AssemblyBasedCallerGenotypingEngine::prepareReadAlleleLikelihoodsForAnnotation` except for the following lines that are missing in GATK 4:

```java
if (call.getAlleles().size() != readAlleleLikelihoodsForAnnotations.alleleCount()) {
  readAlleleLikelihoodsForAnnotations.updateNonRefAlleleLikelihoods(new IndexedAlleleList<>(new HashSet<>(call.getAlleles())));
}
```

I can't think of a reason for this code to be removed, so why don't we restore those lines to `prepareReadAlleleLikelihoodsForAnnotation` and copy the GATK 3 code for `ReadLikelihoods. updateNonRefAlleleLikelihoods()`, which doesn't seem to exist or have any equivalent in GATK 4?

**Comment by droazen** [comment]
Sounds like a good plan to me @davidbenjamin !

**Comment by davidbenjamin** [comment]
@droazen would you like me to take care of that by making a branch for @meganshand to test?

**Comment by droazen** [comment]
@davidbenjamin If you have the time to spare, by all means! Would save @asmirnov239 some work. Feel free to submit a PR against GATK public with this change, plus any unit test, etc., you feel is appropriate. Thanks!
